qutebrowser/browser/hints.py

```diff
@@ -570,23 +570,27 @@ def _check_args(self, target, *args):
                 raise cmdutils.CommandError(
                     "'args' is only allowed with target userscript/spawn.")
 
-    def _filter_matches(self, filterstr, elemstr):
+    def _filter_matches(self, filterstr: str, elem):
         """Return True if `filterstr` matches `elemstr`."""
         # Empty string and None always match
         if not filterstr:
             return True
         filterstr = filterstr.casefold()
-        elemstr = elemstr.casefold()
-        # Do multi-word matching
+        elemstr = str(elem).casefold()
+        try:
+            title = elem.title().casefold()
+            elemstr += " " + title
```

Comment:
I don't think this is the right place to add this to the elemstr - it would make more sense for people to pass this function the extended list. Even better, a list of words to match on.

I'm not sure where best to do that though. Maybe a seperate function to generate an elemstr from an element would be useful.

Reply:
What do you mean by "a list of words to match on" ?

If you mwan which attributes of the element to match against, then the list is pretty limited, we have `title` and `alt`.

Another solution to the issue, would be to modify the method that converts an element to string to include these attributes if they exist. 

Let me know which one would be preferred.

Reply:
I don't mean a list of elements, but a list of the attributes. Eg: if an element's title is "my-title" and alt is "my-alt", I would expect ["my-title", "my-alt"] to be passed in.

I don't really care about a list vs a string though, I just don't think `elemstr` should be touched in this function.

Reply:
Please take a look at the updated code, this diff doesn't correspond to the code any more. The current version should match what you mentioned.